### PR TITLE
[MI-3485] Fix content-type of APIs and remove additional information from logs

### PR DESCRIPTION
**Pull request by ayusht2810**
#### Summary
Fix content-type of APIs and remove additional information from logs

### CODE REVIEW

File: server/handlers/handlers.go [hunk 1]
```diff
@@ -255,11 +245,11 @@ func (ah *ActivityHandler) handleCreatedActivity(activityIds msteams.ActivityIds
 	newPost, appErr := ah.plugin.GetAPI().CreatePost(post)
 
 	if appErr != nil {
-		ah.plugin.GetAPI().LogError("Unable to create post", "post", post, "error", appErr)
+		ah.plugin.GetAPI().LogError("Unable to create post", "error", appErr)
```
Comment (raghavaggarwal2308) [inline server/handlers/handlers.go:248]: Just a question. Why are we removing these?

File: server/handlers/handlers.go [hunk 2]
```diff
@@ -242,7 +232,7 @@ func (ah *ActivityHandler) handleCreatedActivity(activityIds msteams.ActivityIds
 	}
 
 	post, errorFound := ah.msgToPost(channelID, senderID, msg, chat)
-	ah.plugin.GetAPI().LogDebug("Post generated", "post", post)
+	ah.plugin.GetAPI().LogDebug("Post generated")
```
Comment (manojmalik20) [inline server/handlers/handlers.go:235]: Can we log post ID only?

**Review by manojmalik20 — COMMENTED** [review]
Please fix the CI. And also check once more if we're logging the message content anywhere.

File: server/handlers/handlers.go (hunk 1 above)
Comment (ayusht2810) [reply server/handlers/handlers.go:248]: these logs contain post details like messages, which we don't want to log

File: server/handlers/handlers_test.go [hunk 3]
```diff
@@ -260,7 +260,7 @@ func TestHandleCreatedActivity(t *testing.T) {
 			setupAPI: func(mockAPI *plugintest.API) {
 				mockAPI.On("GetDirectChannel", "mockUserID-1", "mockUserID-2").Return(&model.Channel{Id: testutils.GetChannelID()}, nil).Times(1)
 				mockAPI.On("GetUser", testutils.GetUserID()).Return(testutils.GetUser(model.ChannelAdminRoleId, "test@test.com"), nil).Once()
-				mockAPI.On("LogDebug", "Post generated").Times(1)
+				mockAPI.On("LogDebug", "Post generated", "PostID", mock.AnythingOfType("string")).Times(1)
```
Comment (manojmalik20) [inline server/handlers/handlers_test.go:263]: Why are you using mock.AnythingOfType? You should use this only when you can't use the ID here.

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**